testing/skia_gold_client/lib/src/release_version.dart

```diff
@@ -0,0 +1,126 @@
+import 'dart:convert';
+
+import 'package:meta/meta.dart';
+
+/// Signals that the output of golden files may diverge from `main`.
+///
+/// By default, Skia Gold will compare the output of a test to the golden file
+/// in the `main` branch. During a release branch, the output of the test may
+/// diverge (intentionally or accidentally, i.e. a bug) from the output in
+/// `main`.
+///
+/// This class signals to Skia Gold to use a unique test name for the release.
+@immutable
+final class ReleaseVersion {
+  /// Creates a [ReleaseVersion] with the given [major] and [minor] version.
+  ///
+  /// For example, for the `3.21` release:
+  /// ```dart
+  /// ReleaseVersion(
+  ///   major: 3,
+  ///   minor: 21,
+  /// )
+  /// ```
+  ///
+  /// Each number must be non-negative.
+  ReleaseVersion({
+    required this.major,
+    required this.minor,
+  }) {
+    RangeError.checkNotNegative(major, 'major');
+    RangeError.checkNotNegative(minor, 'minor');
+  }
+
+  /// Parses a [ReleaseVersion] from the contents of a `release.version` file.
+  ///
+  /// Returns `null` if and only if the version is parsed as the string `none`.
+  ///
+  /// The format of the file is plaintext, and any lines that are empty
+  /// (newline characters only) or start with `#` are ignored. The first line
+  /// that is not ignored must be in the format `major.minor`, where `major` and
+  /// `minor` are non-negative integers.
+  ///
+  /// For example, the following file contents:
+  /// ```txt
+  /// # This is a comment
+  ///
+  /// 3.21
+  /// ```
+  ///
+  /// ... would parse to `ReleaseVersion(major: 3, minior: 21)`.
+  ///
+  /// Throws a [FormatException] if the file contents are not in the expected
+  /// format (either empty/comments only, or missing a `major.minor` line in
+  /// the format described in [ReleaseVersion.new]).
+  static ReleaseVersion? parse(String fileContents) {
```

Comment:
nit: I think this could be 1/3 the size if you use regular expressions.

Reply:
Hmm. I hope this is fine, I am not opposed to a regular expression but also hate multi-line regular expressions personally.